xapian-applications/omega/handler_libetonyek.cc

```diff
@@ -0,0 +1,233 @@
+	}
+    }
+}
+
+static void
+parse_metadata(const char* data,
+	       size_t len,
+	       string& author,
+	       string& title,
+	       string& keywords)
+{
+    const char* p = data;
+    const char* end = p + len;
+
+    while (p != end) {
+	const char* start = p;
+	p = static_cast<const char*>(memchr(p, '\n', end - start));
+	const char* eol;
+	if (p)
+	    eol = p++;
+	else
+	    p = eol = end;
+	if ((end - start) > 5 && memcmp(start, "meta:", 5) == 0) {
+	    start += 5;
+	    switch (*start) {
+		case 'i': {
+		    if (author.empty())
+			PARSE_FIELD(start, eol, "initial-creator", author);
+		    break;
+		}
+		case 'k': {
+		    PARSE_FIELD(start, eol, "keyword", keywords);
+		    break;
+		}
+	    }
+	} else if ((end - start) > 3 && memcmp(start, "dc:", 3) == 0) {
+	    start += 3;
+	    switch (*start) {
+		case 'c': {
+		    if (!author.empty())
+			author.clear();
+		    PARSE_FIELD(start, eol, "creator", author);
+		    break;
+		}
+		case 's': {
+		    PARSE_FIELD(start, eol, "subject", keywords);
+		    break;
+		}
+		case 't': {
+		    PARSE_FIELD(start, eol, "title", title);
+		    break;
+		}
+	    }
+	} else if ((end - start) > 8 && memcmp(start, "dcterms:", 8) == 0) {
+	    start += 8;
+	    PARSE_FIELD(start, eol, "available", keywords);
+	}
+    }
+}
+
+static void
+clear_text(string& str, const char* text)
+{
+    if (text) {
+	for (int i = 0; text[i] != '\0'; ++i)
+	    if (!isspace(text[i]) || (i && !isspace(text[i - 1])))
+		str.push_back(text[i]);
+	str.push_back('\n');
+    }
+}
+
+static bool
+extract_key(string& dump, RVNGInputStream* input, string& error)
+{
+    RVNGStringVector content;
+    RVNGTextPresentationGenerator document(content);
+    if (!EtonyekDocument::parse(input, &document)) {
+	error = "Libetonyek Error: Fail to extract text";
+	return false;
+    }
+    // Extract text if possible
+    unsigned size = content.size();
+    for (unsigned i = 0; i < size; ++i)
+	clear_text(dump, content[i].cstr());
+    return true;
+}
+
+static bool
+extract_numbers(string& dump, RVNGInputStream* input, string& error)
+{
+    RVNGStringVector content;
+    RVNGTextSpreadsheetGenerator document(content);
+
+    if (!EtonyekDocument::parse(input, &document)) {
+	error = "Libetonyek Error: Fail to extract text";
+	return false;
+    }
+    // Extract text if possible
+    unsigned size = content.size();
+    for (unsigned i = 0; i < size; ++i)
+	clear_text(dump, content[i].cstr());
+    return true;
+}
+
+static bool
+extract_pages(string& dump,
+	      string& title,
+	      string& author,
+	      string& keywords,
+	      string& error,
+	      RVNGInputStream* input)
+{
+    RVNGString text_dump, data_dump;
+    bool succeed = false;
+
+    // Extract text if possible
+    RVNGTextTextGenerator content(text_dump, false);
+    if (EtonyekDocument::parse(input, &content)) {
+	clear_text(dump, text_dump.cstr());
+	succeed = true;
+    } else {
+	 error = "Libetonyek Error: Fail to extract text";
```

Comment:
There's an extra space in the indent here.

Reply:
Sorry, I didn't see it.